src/applications/edu-benefits/0994/submit-transformer.js

```diff
@@ -1,10 +1,60 @@
+import _ from '../../../platform/utilities/data';
 import { transformForSubmit } from 'us-forms-system/lib/js/helpers';
 
 export function transform(formConfig, form) {
-  const formData = transformForSubmit(formConfig, form);
+  const usFormTransform = () =>
+    JSON.parse(transformForSubmit(formConfig, form));
+
+  const addPhoneAndEmail = formData => {
+    const clonedData = _.cloneDeep(formData);
+    if (form.data['view:phoneAndEmail']) {
+      const { dayTimePhone, nightTimePhone, emailAddress } = form.data[
+        'view:phoneAndEmail'
+      ];
+
+      return {
+        ...clonedData,
+        dayTimePhone,
+        nightTimePhone,
+        emailAddress,
+      };
+    }
+    return clonedData;
+  };
+
+  const transformHighTechnologyEmploymentType = formData => {
+    const clonedData = _.cloneDeep(formData);
+    if (form.data['view:salaryEmploymentTypes']) {
+      const { currentSalary, highTechnologyEmploymentType } = form.data[
+        'view:salaryEmploymentTypes'
+      ];
+
+      const highTechnologyEmploymentTypes = Object.keys(
+        highTechnologyEmploymentType,
+      )
+        .filter(key => highTechnologyEmploymentType[key])
+        .map(key => key);
+
+      delete clonedData.highTechnologyEmploymentType;
```

Comment:
not familiar with this data structure but this might need to delete `clonedData['view:salaryEmploymentTypes'].highTechnologyEmploymentType`. not clear to me why it would be on that view and on the root of the `formData`.

Reply:
When `us-forms-system/lib/js/helpers#transformForSubmit` calls `filterViewFields` it pulls up any non-view field properties from the viewField. If I remove the delete submit-transformer unit test fails because `educationBenefits.form.highTechnologyEmploymentType` exists when it shouldn't after transform.